src/compile/mark/rule.ts

```diff
@@ -12,25 +12,34 @@ export namespace rule {
     let p: any = {};
 
     // TODO: support explicit value
+    const orient = model.config().mark.orient;
 
-    // vertical
-    if (model.has(X)) {
-      p.x = position(model, X);
-
-      p.y = { value: 0 };
-      p.y2 = {
-          field: {group: 'height'}
+    if (orient === 'vertical') {
+      if (model.has(X) && model.has(Y) && model.has(Y2)) {
```

Comment:
Why don't you split x and y inside this case?  

- You haven't handle the case 
  - where there is y2, but no y.  set `y` to `{group: 'height'}`)
  - both `y` and `y2` are not specified
  - where there is no x.  (Returns `{value: 0}`)